app/frontend/src/features/search/SearchPage.tsx

```diff
@@ -113,16 +117,34 @@ export default function SearchPage() {
       }
       addUsersToMap(newMap, handleMapUserClick);
     });
+    newMap.on("click", (e) => {
+      if (!e.defaultPrevented) {
+        handleResultClick(undefined);
+      }
```

Comment:
Does this only get registered once when the map is initialised? i.e. in subsequent renders, this will continue to be used until the map is destroyed? That might also explain the outdated state the callback gets as it only keeps seeing the state value from the closure it's created in 🤔